### PR TITLE
Story/autoneal combustible land area/2193

### PR DESCRIPTION
**Pull request by andrea-williams**
Example (running locally):

Fraser Fire Zone is now under advisory because >= 10% of its combustible area has high HFI.

Screen Shot 2022-09-12 at 4 25 27 PM

### CODE REVIEW

File: api/app/wildfire_one/query_builders.py [hunk 1]
```diff
@@ -26,7 +26,9 @@ def __init__(self):
         super().__init__()
         self.param_query = None
         # In conversation with Dana Hicks, on Apr 20, 2021 - Dana said to show active, test and project.
-        for status in ('ACTIVE', 'TEST', 'PROJECT'):
+        # Added Disabled status to fix situation where stations Disabled in WFWX API are being used in
+        # HFI Calc, causing Planning Area-level calculations to break due to missing station data.
+        for status in ('ACTIVE', 'TEST', 'PROJECT', 'DISABLED'):
```
Comment (Sybrand) [inline api/app/wildfire_one/query_builders.py:31]: This is going to hang around for quite a while in classify_hfi - shouldn't we have this go into a separate PR?

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 2]
```diff
@@ -63,7 +63,6 @@ export interface FBAMapProps {
 export const hfiSourceFactory = (url: string) => {
   return new XYZ({
     url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=${url}&source=hfi`,
-    interpolate: false,
```
Comment (Sybrand) [inline web/src/features/fba/components/map/FBAMap.tsx:66]: Why are you removing this?

Comment (andrea-williams) [reply web/src/features/fba/components/map/FBAMap.tsx:66]: Couldn't tell ya. I've merged Conor's branch into mine, and that `interpolate` change has come from his branch.

File: api/advisory/db/crud.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+from sqlalchemy import select
+from sqlalchemy.engine.row import Row
+from sqlalchemy.ext.asyncio import AsyncSession
+from advisory.db.models.tileserver import (Hfi, FireZone)
+
+logger = logging.getLogger(__name__)
+
+
+async def get_hfi_area_percentages(session: AsyncSession, for_date: date) -> List[Row]:
+    """ DEPRECATED?
```
Comment (Sybrand) [inline api/advisory/db/crud.py:15]: Yes - pls. remove this file. It's not used by anything anymore.

File: api/advisory/db/models/tileserver.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 """ We don't want alembic to manage the models in here!
 """
 
-from sqlalchemy import (Column, Integer, Date, String)
+from sqlalchemy import (Column, Integer, Date, String, Float)
```
Comment (Sybrand) [inline api/advisory/db/models/tileserver.py:4]: No need to touch this model - we should remove this model entirely, since there's no need to touch the tileserver database from the API.

File: api/advisory/db/crud.py (hunk 3 above)
Comment (andrea-williams) [reply api/advisory/db/crud.py:15]: 😆 you were the one who added the Deprecated? comment.

But yes, I'll delete this file.

File: web/src/features/fba/components/map/FBAMap.tsx (hunk 2 above)
Comment (Sybrand) [reply web/src/features/fba/components/map/FBAMap.tsx:66]: He hadn't run yarn update - and removed that to make it work. Can you please roll back the change?

File: api/app/wildfire_one/query_builders.py (hunk 1 above)
Comment (andrea-williams) [reply api/app/wildfire_one/query_builders.py:31]: That was a residual file change from ages ago - it snuck into this PR. Removing it from the PR

File: api/advisory/polygonize_geotiff.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import sys
```
Comment (Sybrand) [inline api/advisory/polygonize_geotiff.py:1]: I'd like this function to move to api/app/auto_spatial_advisory/polygonize.py - I think splitting out into api/advisory was a mistake (my bad - premature separation of code?).

Can you please move it there? Side by side with the other `polygonize` function it's also very clear what re-factoring and renaming we're going to have to do.

File: api/advisory/transform_to_epsg_3005.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
+import sys
```
Comment (Sybrand) [inline api/advisory/transform_to_epsg_3005.py:1]: This feels like a generalize geospatial utility. What do you think about throwing it into api/app/utils ?

**Review by Sybrand — COMMENTED** [review]
@andrea-williams I have a bunch of comments for you - let me know if you want to chat.

File: api/advisory/db/crud.py (hunk 3 above)
Comment (Sybrand) [reply api/advisory/db/crud.py]: Yeah - deprecated and deleted :) That file has been removed, so you'd be re-introducing it.

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 7]
```diff
@@ -72,7 +72,6 @@ export const hfiSourceFactory = (url: string) => {
 export const ftlSourceFactory = (filter: string) => {
   return new XYZ({
     url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/ftl/ftl_2018_cloudoptimized.tif&source=ftl&filter=${filter}`,
-    interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fba/components/map/FBAMap.tsx:75]: Please revert this change.

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 8]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
 
 export const ftlSource = new XYZ({
   url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/ftl/ftl_2018_cloudoptimized.tif&source=ftl`,
-  interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/maps/constants.ts:26]: Please revert this change.

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 9]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
...
   tileSize: COG_TILE_SIZE
 })
 
 export const sfmsFtlSource = new XYZ({
   url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/sfms/cog/static/cog_fbp2021.tif&source=ftl`,
-  interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/maps/constants.ts:32]: Please revert this change.

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 10]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
...
   tileSize: COG_TILE_SIZE,
   maxZoom: SFMS_MAX_ZOOM
 })
 
 export const sfmsElevationSource = new XYZ({
   url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/sfms/cog/static/cog_bc_elevation.tif&source=elevation`,
-  interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/maps/constants.ts:39]: Please revert this change.

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 11]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
...
   tileSize: COG_TILE_SIZE,
   maxZoom: SFMS_MAX_ZOOM
 })
 
 export const twelveArcElevationSource = new XYZ({
   url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/dem/cog/cog_BC_Area_CDEM.tif&source=elevation`,
-  interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/maps/constants.ts:46]: Please revert this change.

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 12]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
...
   tileSize: COG_TILE_SIZE
 })
 
 export const sfmsSlopeSource = new XYZ({
   url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/sfms/cog/static/cog_bc_slope.tif&source=slope`,
-  interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/maps/constants.ts:52]: Please revert this change.

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 13]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
...
   tileSize: COG_TILE_SIZE,
   maxZoom: SFMS_MAX_ZOOM
 })
 
 export const sfmsAspectSource = new XYZ({
   url: `${RASTER_SERVER_BASE_URL}/tile/{z}/{x}/{y}?path=gpdqha/sfms/cog/static/cog_bc_aspect.tif&source=aspect`,
-  interpolate: true,
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/maps/constants.ts:59]: Please revert this change.

File: api/app/auto_spatial_advisory/calculate_combustible_land_area.py [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+async def calculate_combustible_area_by_fire_zone() -> List[Dict[int, float]]:
+    # Simplified fuel type layer that SFMS system uses (fbp2021.tif) has been converted to EPSG:3005,
+    # polygonized, and saved to our object store bucket.
+    gdal.SetConfigOption('AWS_SECRET_ACCESS_KEY', config.get('OBJECT_STORE_SECRET'))
+    gdal.SetConfigOption('AWS_ACCESS_KEY_ID', config.get('OBJECT_STORE_USER_ID'))
+    gdal.SetConfigOption('AWS_S3_ENDPOINT', config.get('OBJECT_STORE_SERVER'))
+    gdal.SetConfigOption('AWS_VIRTUAL_HOSTING', 'FALSE')
+    bucket = config.get('OBJECT_STORE_BUCKET')
+    fuel_types_vector_filepath = f'/vsis3/{bucket}/ftl/fuel_types_epsg_3005.shp'
```
Comment (Sybrand) [inline api/app/auto_spatial_advisory/calculate_combustible_land_area.py:31]: Can we give this file a name that makes it clear where it's from? I'm assuming it's from the SMFS tiff somehow? But that's not clear.

Also - we need the same file in the production bucket, for when this code get's merged.

File: web/src/features/fba/components/map/FBAMap.tsx (hunk 7 above)
Comment (conbrad) [inline web/src/features/fba/components/map/FBAMap.tsx:75]: Unnecessary (sorry)

File: web/src/features/fireWeather/components/maps/constants.ts [hunk 15]
```diff
@@ -23,40 +23,34 @@ export const monochromeSource = new XYZ({
 
 export const ftlSource = new XYZ({
```
Comment (conbrad) [inline web/src/features/fireWeather/components/maps/constants.ts:24]: All removals of `interpolate` unnecessary (sorry)

**Review by conbrad — APPROVED** [review]
LGTM, just need to address the other `interpolate` removal instanaces.

**Review by Sybrand — COMMENTED** [review]
Having trouble running code.... more comments to come

File: api/alembic/versions/d9c05cb16869_add_combustible_area_data_to_advisory_.py [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+revision = 'd9c05cb16869'
+down_revision = '0669994d4089'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    zone_areas = loop.run_until_complete(calculate_combustible_area_by_fire_zone())
```
Comment (Sybrand) [inline api/alembic/versions/d9c05cb16869_add_combustible_area_data_to_advisory_.py:23]: This won't work - you're opening a new session in calculate_combustible_area_by_fire_zone which is going to be blocked by the transaction alembic has started.

**Review by Sybrand — CHANGES_REQUESTED** [review]

File: api/app/auto_spatial_advisory/calculate_combustible_land_area.py (hunk 14 above)
Comment (andrea-williams) [reply api/app/auto_spatial_advisory/calculate_combustible_land_area.py:38]: Renamed to `fuel_types_from_sfms_epsg_3005` in dev and prod buckets

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 17]
```diff
@@ -72,7 +72,7 @@ export const hfiSourceFactory = (url: string) => {
...
+    interpolate: false,
```
Comment (Sybrand) [inline web/src/features/fba/components/map/FBAMap.tsx:75]: Why is this false now?

Comment (andrea-williams) [reply web/src/features/fba/components/map/FBAMap.tsx]: I made a boo-boo. Switched back to true

**Review by Sybrand — APPROVED** [review]
